airbyte-integrations/connectors/destination-databricks/src/main/kotlin/io/airbyte/integrations/destination/databricks/DatabricksDestination.kt

```diff
@@ -0,0 +1,196 @@
+import io.airbyte.cdk.integrations.destination.async.buffers.BufferManager
+import io.airbyte.cdk.integrations.destination.async.deser.AirbyteMessageDeserializer
+import io.airbyte.cdk.integrations.destination.s3.FileUploadFormat
+import io.airbyte.cdk.integrations.util.addDefaultNamespaceToStreams
+import io.airbyte.integrations.base.destination.typing_deduping.CatalogParser
+import io.airbyte.integrations.base.destination.typing_deduping.StreamId
+import io.airbyte.integrations.destination.databricks.jdbc.DatabricksDestinationHandler
+import io.airbyte.integrations.destination.databricks.jdbc.DatabricksSqlGenerator
+import io.airbyte.integrations.destination.databricks.jdbc.DatabricksStorageOperations
+import io.airbyte.integrations.destination.databricks.model.DatabricksConnectorConfig
+import io.airbyte.integrations.destination.databricks.staging.DatabricksFlushFunction
+import io.airbyte.integrations.destination.databricks.staging.SerializableBufferFactory
+import io.airbyte.integrations.destination.databricks.sync.DatabricksSyncOperations
+import io.airbyte.protocol.models.v0.AirbyteConnectionStatus
+import io.airbyte.protocol.models.v0.AirbyteMessage
+import io.airbyte.protocol.models.v0.ConfiguredAirbyteCatalog
+import io.airbyte.protocol.models.v0.DestinationSyncMode
+import io.github.oshai.kotlinlogging.KotlinLogging
+import java.util.*
+import java.util.function.Consumer
+
+private val log = KotlinLogging.logger {}
+
+class DatabricksDestination : BaseConnector(), Destination {
+    override fun getConsumer(
+        config: JsonNode,
+        catalog: ConfiguredAirbyteCatalog,
+        outputRecordCollector: Consumer<AirbyteMessage>
+    ): AirbyteMessageConsumer? {
+        throw UnsupportedOperationException(
+            "GetConsumer is not supported, use getSerializedMessageConsumer"
+        )
+    }
+
+    override fun check(config: JsonNode): AirbyteConnectionStatus? {
+        // TODO: Add proper checks for
+        //   Check schema permissions, or if raw_override and default already exists
+        //   Check catalog permissions to USE catalog
+        //   Check CREATE volume, COPY INTO, File upload permissions
+        //   Check Table creation, Table drop permissions
+        val connectorConfig = DatabricksConnectorConfig.deserialize(config)
+        val sqlGenerator =
+            DatabricksSqlGenerator(DatabricksNamingTransformer(), connectorConfig.database)
+        val datasource = ConnectorClientsFactory.createDataSource(connectorConfig)
+        val jdbcDatabase = DefaultJdbcDatabase(datasource)
+        val destinationHandler =
+            DatabricksDestinationHandler(connectorConfig.database, jdbcDatabase)
+        val workspaceClient =
+            ConnectorClientsFactory.createWorkspaceClient(
+                connectorConfig.hostname,
+                connectorConfig.authentication
+            )
+        val storageOperations =
+            DatabricksStorageOperations(
```

Comment:
dumb question: is there any case where we'll use the `DatabricksStorageOperations` with anything else than those particular types? Should we just create them inside that constructor?

Reply:
injected as dependency for mocking them, although didn't write unit test in this 😅 . Will do that in CDK PR when doing the bigquery thing now.

Reply:
Yeah, if we can eventually fold our SqlGen tests as StorageOps tests. TBH it needs for injection are ConnectorConfig static info like schema, catalog, jdbcurl and both SqlGen and DestHandler can be internally instantiated.